src/Widgets/StripeDialog.vala

```diff
@@ -0,0 +1,381 @@
+/*
+* Copyright (c) 2016 elementary LLC (https://launchpad.net/granite)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 2 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+* Boston, MA 02111-1307, USA.
+*
+*/
+
+public class AppCenter.Widgets.StripeDialog : Gtk.Dialog {
+    private const string HOUSTON_URI =  "https://developer.elementary.io/api/payment/%s?key=%s&token=%s&amount=%s&currency=USD";
+    private const string USER_AGENT = "Stripe checkout";
+    private const string STRIPE_URI = "https://api.stripe.com/v1/tokens?email=%s"
+                            + "&payment_user_agent=%s&amount=%s&card[number]=%s"
+                            + "&card[cvc]=%s&card[exp_month]=%s&card[exp_year]=%s"
+                            + "&key=%s";
+
+    private Gtk.Grid card_layout;
+    private Gtk.Grid? processing_layout = null;
+    private Gtk.Grid? error_layout = null;
+    private Gtk.Stack layouts;
+
+    private Gtk.Entry email_entry;
+    private Gtk.Entry card_number_entry;
+    private Gtk.Entry card_expiration_entry;
+    private Gtk.Entry card_cvc_entry;
+    private Gtk.Button pay_button;
+    private Gtk.Button cancel_button;
+
+    public int amount { get; construct set; }
+    public string app_name { get; construct set; }
+    public string app_id { get; construct set; }
+    public string stripe_key { get; construct set; }
+
+    private bool email_valid = false;
+    private bool card_valid = false;
+    private bool expiration_valid = false;
+    private bool cvc_valid = false;
+
+    public StripeDialog (int amount_, string app_name_, string app_id_, string stripe_key_) {
+        Object (amount: amount_, app_name: app_name_, app_id: app_id_, stripe_key: stripe_key_);
+        deletable = false;
+        resizable = false;
+
+        var primary_label = new Gtk.Label ("AppCenter");
+        primary_label.get_style_context ().add_class ("primary");
+
+        var secondary_label = new Gtk.Label (app_name);
+
+        email_entry = new Gtk.Entry ();
+        email_entry.input_purpose = Gtk.InputPurpose.EMAIL;
+        email_entry.placeholder_text = "Email";
+        email_entry.primary_icon_name = "internet-mail-symbolic";
+
+        email_entry.changed.connect (() => {
+           email_entry.text = email_entry.text.replace (" ", "").down ();
+           validate (0, email_entry.text);
```

Comment:
It would be better to create some kind of enum to describe what are you trying to validate.

Like:
`public enum EntryType {
    EMAIL,
    CARD_NUMBER,
    CARD_EXP,
    CARD_CVC
}`